src/Form/Form.js

```diff
@@ -103,14 +109,7 @@ class Form {
             }
         )
 
-        // This ensures we fire the handler again if the form is changed
-        this.addListener(form, 'input', () => {
-            if (!this.isAutofilling) {
-                this.handlerExecuted = false
-                this.shouldPromptToStoreData = true
-            }
-        })
-
+        this.initFormListeners()
```

Comment:
Moved to initialising the listeners in a central form. This will make it easier in the future to delegate event listeners at least at the form level, instead of attaching to every single input field.